tests/routes/adage_iframe/adage_iframe_test.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+
+import jwt
+
+from pcapi import settings
+from pcapi.core.users.models import ALGORITHM_RS_256
+
+from tests.conftest import TestClient
+
+
+class Returns200Test:
+    def test_should_return_success_response_when_jwt_valid(self, app):
+        # Given
+        now = datetime.datetime.utcnow()
+        test_client = TestClient(app.test_client())
+        with open(settings.VALID_RSA_PRIVATE_KEY_PATH, "rb") as reader:
+            valid_encoded_token = jwt.encode(
+                {"email": "jean@test.com", "exp": now + datetime.timedelta(days=1)},
+                reader.read(),
+                algorithm=ALGORITHM_RS_256,
+            )
+
+        test_client.auth_header = {"Authorization": f"Bearer {valid_encoded_token}"}
+
+        # When
+        response = test_client.get("/adage-iframe/")
+
+        # Then
+        assert response.status_code == 200
+
+
+class ReturnsErrorTest:
+    def test_should_return_error_response_when_jwt_invalid(self, app):
+        # Given
+        now = datetime.datetime.utcnow()
+        test_client = TestClient(app.test_client())
+        with open(settings.INVALID_RSA_PRIVATE_KEY_PATH, "rb") as reader:
+            corrupted_token = jwt.encode(
+                {"email": "jean@test.com", "exp": now + datetime.timedelta(days=1)},
+                reader.read(),
+                algorithm=ALGORITHM_RS_256,
+            )
+
+        test_client.auth_header = {"Authorization": f"Bearer {corrupted_token}"}
+
+        # When
+        response = test_client.get("/adage-iframe/")
+
+        # Then
+        assert response.status_code == 403
+        assert "Unrecognized token" in response.json["Authorization"]
+
+    def test_should_return_error_response_when_jwt_expired(self, app):
+        # Given
+        now = datetime.datetime.utcnow()
+        test_client = TestClient(app.test_client())
+        with open(settings.VALID_RSA_PRIVATE_KEY_PATH, "rb") as reader:
+            valid_token = jwt.encode(
+                {"email": "jean@test.com", "exp": now - datetime.timedelta(days=1)},
+                reader.read(),
+                algorithm=ALGORITHM_RS_256,
+            )
+
+        test_client.auth_header = {"Authorization": f"Bearer {valid_token}"}
+
+        # When
+        response = test_client.get("/adage-iframe/")
+
+        # Then
+        assert response.status_code == 422
+        assert "Token expired" in response.json["msg"]
+
+    def test_should_return_error_response_when_no_expiration_date_in_token(self, app):
+        # Given
+        test_client = TestClient(app.test_client())
+        with open(settings.VALID_RSA_PRIVATE_KEY_PATH, "rb") as reader:
+            valid_token = jwt.encode(
```

Comment:
```suggestion
            no_expiration_date_token = jwt.encode(
```
+ changer aussi sur la l83